src/actions/notification.ts

```diff
@@ -0,0 +1,48 @@
+import { StreamElement, TurboStreamActions } from "@hotwired/turbo"
+import { camelize, typecast } from "../utils"
+
+const PERMITTED_ATTRIBUTES = [
+  "dir",
+  "lang",
+  "badge",
+  "body",
+  "tag",
+  "icon",
+  "image",
+  "data",
+  "vibrate",
+  "renotify",
+  "require-interaction",
+  "actions",
+  "silent",
+]
+
+const createNotification = (streamElement: StreamElement) => {
+  const title = streamElement.getAttribute("title") || ""
+
+  const options = Array.from(streamElement.attributes)
+    .filter((attribute) => PERMITTED_ATTRIBUTES.includes(attribute.name))
+    .reduce((acc, attribute) => {
+      return { ...acc, [camelize(attribute.name)]: typecast(attribute.value) }
+    }, {})
```

Comment:
Personally, I think `Object.fromEntries(...)` prepended with a `.map(...)` is a lot easier to understand than a `.reduce(...)`
```suggestion
  const attributes = Array.from(streamElement.attributes)
    .filter((attribute) => PERMITTED_ATTRIBUTES.includes(attribute.name))
    .map((attribute) => [camelize(attribute.name), typecast(attribute.value))

  const options = Object.fromEntries(attributes)
```

Reply:
Very nice, thank you!